DOC-2975: Added new page 'Load from Apache Iceberg' [4.3]

Tushar-TG-14 opened the pull request.

Tushar-TG-14 committed:
DOC-2975: Added new page 'Load from Apache Iceberg' [4.3]

pingxieTG reviewed (commented):
Overall lgtm. But there are some partials that can be referenced and reused, e.g., https://github.com/tigergraph/server-docs/blob/4.2/modules/data-loading/pages/load-from-cloud.adoc?plain=1#L32

victorleeTG commented on modules/data-loading/pages/load-from-iceberg.adoc:
```diff
@@ -1 +1,306 @@
+:toclevels: 4
 = Load from Apache Iceberg
+
+In version *4.3*, TigerGraph introduces a connector to load data from an *Apache Iceberg*. This connector allows you to load data stored in *AWS S3* or *MinIO* buckets that are managed by an *Iceberg REST Catalog*.
```
```suggestion
TigerGraph 4.3 adds *Apache Iceberg* to its collection of high-speed built-in connectors. This connector allows you to load data stored in *AWS S3* or *MinIO* buckets that are managed by an *Iceberg REST Catalog*.
```
Reasons for the change:
1. We want to emphasize that we are expanded an existing family. not adding something that is totally new. Writing our documentation as though everything is new is one reason why our documentation is difficult to read: it's written like a large number of individual features, rather than as a smooth and logical fabric.
2. Avoid talking about TigerGraph the company. Talk about TigerGraph DB the product. Notice how the meaning of "TigerGraph" is different in the two versions. Is this were marketing material, talking about the company would be fine, but this is technical documentation, not marketing.
3. We do not say "an Iceberg", just like we do not say "a Spark".

victorleeTG commented on modules/data-loading/pages/load-from-iceberg.adoc:
```diff
@@ -1 +1,306 @@
+
+This guide shows how to connect your data source, create a loading job, and manage it effectively.
+
+== Build Your Graph Foundation
```
```suggestion
== Build Your Graph Schema
```
Both TigerGraph and Iceberg use the term "schema". "Foundation" is not a standard term for either product, so why introduce a new concept that isn't needed?

victorleeTG commented on modules/data-loading/pages/load-from-iceberg.adoc:
```diff
@@ -1 +1,306 @@
+  aws.s3.access_key: admin,
+  aws.s3.secret_key: password,
+  aws.client.region: us-east-1,
+  tasks.max: 2
```
In our current documentation, `tasks.max` is a filename parameter and can only be configured when defining a `FILENAME`.  
Is the ability to define `tasks.max` in a `DATA SOURCE` a new or old feature? Does it apply to all connectors or just Iceberg?
Does it apply to all filename parameters or just one (or a few)? Which ones?

This is the problem with documenting by "example". You actually create more questions than you answer. The better approach is to briefly explain the feature (or option) and then show an example. Or, show an example and then explain it. At some point, provide all the important details for what is/isn't supported.

victorleeTG commented on modules/data-loading/pages/load-from-iceberg.adoc:
```diff
@@ -1 +1,306 @@
+  LOAD f1 TO VERTEX person VALUES ($0, $1, $2);
+}
+
+- *8 Tasks from Job*: Use 8 tasks for larger data.
```
Why these numbers, 2 and 8?

1 is the default, right? Is there a global max?

victorleeTG commented on modules/data-loading/pages/load-from-iceberg.adoc:
```diff
@@ -1 +1,306 @@
+
+[source,gsql]
+CREATE LOADING JOB loadSocialNet FOR GRAPH socialNet {
+  DEFINE FILENAME f1 = "$s1:SELECT personId, id, gender FROM iceberg_connector.person WHERE gender = 'male'";
```
Is `iceberg_connector` a built-in name or this a user-defined name? If it was user defined, where/when would it be defined?

victorleeTG commented on modules/data-loading/pages/load-from-iceberg.adoc:
```diff
@@ -1 +1,306 @@
+
+Create a loading job to turn Iceberg data into your graph’s vertices and edges. It involves defining data sources and mapping the data.
+
+=== Quick Loading Job Example
```
You should include the Iceberg schema. A user cannot really learn from a loading example if you are only showing the schema of one half (either the source schema or the target graph schema). We need to see both schemas.

victorleeTG commented on modules/data-loading/pages/load-from-iceberg.adoc:
```diff
@@ -1 +1,306 @@
+Connect to MinIO with a REST catalog:
+
+[source,gsql]
+CREATE DATA_SOURCE s1 = """
+{
+    type: iceberg,
+    iceberg.catalog.type: rest,
+    iceberg.catalog.uri: http://rest:8181,
+    aws.s3.endpoint: http://minio:9000,
+    aws.s3.access_key: accesskey,
+    aws.s3.secret_key: password,
+    aws.client.region: us-east-1
+}""" FOR GRAPH socialNet
```
Other TigerGraph connectors let you put this configuration JSON in a file. I assume that is also supported here?

If this is in a file, I suspect the triple quotes are omitted, but I'm not sure.

Tushar-TG-14 replied:
@pingxieTG Could you please clarify this point?

pingxieTG replied:
Yes it supports JSON file as well. It is no different from the way other data sources are created. @Tushar-TG-14

victorleeTG commented on modules/data-loading/pages/load-from-iceberg.adoc:
```diff
@@ -1 +1,306 @@
+  LOAD f1 TO VERTEX person VALUES ($0, $1, $2);
+}
+
+== Define Your Data Files
```
This section confused me because it was repeating things that we're shown in the examples above.
After reading further, I eventually realized that you were breaking down the full loading process, step-by-step, which was all lumped together in the examples before.

Please add introductory or transitional sentences, and some sort of main heading like "Connector Setup and Loading - Step by Step" to tell the user how you are leading them. Otherwise, the reader doesn't understand how the sections are related to one another.

victorleeTG commented on modules/data-loading/pages/load-from-iceberg.adoc:
```diff
@@ -1 +1,306 @@
+[source,gsql]
+DEFINE FILENAME query_person = "$s1:SELECT personId, id, gender FROM iceberg_connector.person";
+DEFINE FILENAME bq_inline_json = "$s1:myfile.json";
+DEFINE FILENAME query_person = "$s1:{query: 'SELECT personId, id, gender FROM iceberg_connector.person WHERE gender = 'male'', tasks.max: 2}";
```
This example uses a a file object name `query_person` which was already used above. It should be changed to be unique.

victorleeTG commented on modules/data-loading/pages/load-from-iceberg.adoc:
```diff
@@ -1 +1,306 @@
+
+== Connect Your Data Source
+
+Configure a data source object to connect TigerGraph to your Apache Iceberg storage (S3 or MinIO). This involves specifying connection details using JSON.
```
"Using JSON" is very vague. If you say that, I expect some explanation pretty quickly; otherwise you leave me hanging.  So, it's actually better no to say that yet.

The structure  of the existing data connection documentation is logical; it just needs to be streamlined.
The logic:
* **Create a Data Source Object** (instead of Connect Your Data Source)
** **Specify configuration parameters in a JSON object**
*** use a small JSON example to show the format and how it can be specified either inline or in a separate file
*** show the tables of all the configuration parameters

Tushar-TG-14 committed:
DOC-2975: Revise Iceberg loading documentation

Update the documentation for loading data from Apache Iceberg, including configuration parameters, examples, and best practices.
